app/views/csvfiles/new.html.erb

```diff
@@ -0,0 +1,107 @@
+<script src="https://cdnjs.cloudflare.com/ajax/libs/PapaParse/4.6.3/papaparse.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.7.3/Chart.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/plotly.js/1.33.1/plotly-basic.min.js"></script>
+<link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.8.2/css/all.min.css">
+<link href="https://gitcdn.github.io/bootstrap-toggle/2.2.2/css/bootstrap-toggle.min.css" rel="stylesheet">
+<script src="https://gitcdn.github.io/bootstrap-toggle/2.2.2/js/bootstrap-toggle.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/FileSaver.js/1.3.3/FileSaver.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/xlsx/0.14.3/xlsx.full.min.js"></script>
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.8.2/css/all.min.css">
+<%= javascript_include_tag('/lib/simple-data-grapher/dist/PublicLab.Grapher.js')%>
+<%= stylesheet_link_tag '/lib/simple-data-grapher/examples/upload_file.css'%>
+<div id="first"></div>
+<script>
+    SimpleDataGrapherObject = new SimpleDataGrapher("first");
+    var value='<%= current_user %>';
+    <% if current_user %>
+        SimpleDataGrapherObject.view.createButtons("yes");
+        var saveButton=SimpleDataGrapherObject.view.elementId+"_save_CSV";
+        var prevFile=SimpleDataGrapherObject.view.elementId+"_prev_file";
+        var saveFlag=false;
+        $("#"+saveButton).click(function(){
+            saveFlag=true;
+        });
+        $("#"+SimpleDataGrapherObject.view.plotGraphId).click(function(){
+            if (saveFlag){
+                var arr={};
+                arr["completeCsvMatrix"]=SimpleDataGrapherObject.view.csvParser.completeCsvMatrix;
+                arr["csvHeaders"]=SimpleDataGrapherObject.view.csvParser.csvHeaders;
+                arr["csvSampleData"]=SimpleDataGrapherObject.view.csvParser.csvSampleData;
+                arr["csvValidForYAxis"]=SimpleDataGrapherObject.view.csvParser.csvValidForYAxis;
+                arr["completeCsvMatrixTranspose"]=SimpleDataGrapherObject.view.csvParser.completeCsvMatrixTranspose;
+                let csvStringMatrix=SimpleDataGrapherObject.view.csvParser.completeCsvMatrixTranspose;
+                let csvStringForDownload=encodeURI("data:text/csv;charset=utf-8," + csvStringMatrix.map(e => e.join(",")).join("\n"));
+                $.ajax({
+                    url: '/simple-data-grapher/object',
+                    type: 'post',
+                    data:   {object: JSON.stringify(arr), 
+                            uid: <%= current_user.id %>, 
+                            filetitle: SimpleDataGrapherObject.view.fileTitle, 
+                            filedescription: SimpleDataGrapherObject.view.fileDescription, 
+                            filestring: csvStringForDownload},
+                    success: function(data){
```

Comment:
Hi @IshaGupta18 If it's success, then we have data being shown in console. What about having alert maybe in success like `Data Saved`?

Reply:
Hi @IshaGupta18 Js require proper spacing in begin and end of operators. 
Small stuff like space after comma in `Console.log`  would help in better linting.